tests/backup/backup_share_test.go

```diff
@@ -4215,3 +4215,400 @@ var _ = Describe("{SwapShareBackup}", func() {
+									case errCh <- err:
+									default:
+									}
+								}
+							}(backup, ctx)
+							deletedUserBackupMap[backupName] = true
+						case api.BackupInfo_StatusInfo_Pending:
+							pendingBackups = true
+						}
+					} else {
+						log.Warnf("backup %s with uid %s may not have been created by this testcase", backup.GetName(), backup.GetOrgId())
+					}
+				}
+				wg.Wait()
+				select {
+				case err := <-errCh:
+					return "", false, err
+				default:
+					if pendingBackups {
+						return "", true, fmt.Errorf("there are pending backups")
+					}
+					return "", false, nil
+				}
+			}
+			_, err = DoRetryWithTimeoutWithGinkgoRecover(deleteInProgressBackup, maxWaitPeriodForBackupJobCancellation*time.Minute, backupJobCancellationRetryTime*time.Second)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Attempting to delete in progress backups"))
+			dash.VerifyFatal(len(deletedUserBackupMap) > 0, true, "Verifying if there is at least one in progress backup")
+		})
+		Step(fmt.Sprintf("Take failed backup of applications from the user %s", infraAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Taking failed backup of applications from the user %s", infraAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(infraAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", infraAdminUser)
+			var wg sync.WaitGroup
+			createBackup := func(backupName string, namespace string) {
+				defer GinkgoRecover()
+				defer wg.Done()
+				for backupLocationUID, backupLocationName := range userBackupLocationMap {
+					_, err = CreateBackupByNamespacesWithoutCheck(backupName, SourceClusterName, backupLocationName, backupLocationUID,
+						[]string{invalidNamespace}, map[string]string{}, orgID, userSourceClusterUID, "", "", "", "", nonAdminCtx)
+					if err != nil {
+						backupUID, UIDError := Inst().Backup.GetBackupUID(nonAdminCtx, backupName, orgID)
+						log.FailOnError(UIDError, "failed to get backup %s uid", backupName)
+						backupInspectRequest := &api.BackupInspectRequest{
+							Name:  backupName,
+							OrgId: orgID,
+							Uid:   backupUID,
+						}
+						resp, inspectError := Inst().Backup.InspectBackup(nonAdminCtx, backupInspectRequest)
+						log.FailOnError(inspectError, "failed to inspect backup %s", backupName)
+						if resp.GetBackup().GetStatus().GetStatus() != api.BackupInfo_StatusInfo_Failed {
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of backup %s of namespace %s", backupName, namespace))
+						}
+					}
+					break
+				}
+			}
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					backupName := fmt.Sprintf("%s-%s-%d-%v", BackupNamePrefix, namespace, i, time.Now().Unix())
+					wg.Add(1)
+					go createBackup(backupName, namespace)
+					userFailedBackupNames = append(userFailedBackupNames, backupName)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of failed user backups taken are: %v", userFailedBackupNames)
+		})
```

Comment:
I might be missing a step to verify backup by non-admin user is listed from admin pages while backup is in failed state. @ak-px

Reply:
Done.